sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -193,13 +193,32 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
    * This is only used for Common Table Expressions.
    */
   override def visitNamedQuery(ctx: NamedQueryContext): SubqueryAlias = withOrigin(ctx) {
-    val subQuery: LogicalPlan = plan(ctx.query).optionalMap(ctx.columnAliases)(
+    val logicalPlan = Option(ctx.query).map(plan).orElse(
+      Option(ctx.ddlStatementForQuery).map(visitDdlStatementForQuery)).get
+    val subQuery: LogicalPlan = logicalPlan.optionalMap(ctx.columnAliases)(
       (columnAliases, plan) =>
         UnresolvedSubqueryColumnAliases(visitIdentifierList(columnAliases), plan)
     )
     SubqueryAlias(ctx.name.getText, subQuery)
   }
 
+  override def visitDdlQuery(ctx: DdlQueryContext): LogicalPlan = withOrigin(ctx) {
+    visitDdlStatementForQuery(ctx.ddlStatementForQuery())
+  }
+
+  def visitDdlStatementForQuery(ctx: DdlStatementForQueryContext): LogicalPlan = withOrigin(ctx) {
+    ctx match {
+      case namespaces: ShowNamespacesContext => visitShowNamespaces(namespaces)
+      case tables: ShowTablesContext => visitShowTables(tables)
+      case tblProperties: ShowTblPropertiesContext => visitShowTblProperties(tblProperties)
+      case partitions: ShowPartitionsContext => visitShowPartitions(partitions)
+      case columns: ShowColumnsContext => visitShowColumns(columns)
+      case views: ShowViewsContext => visitShowViews(views)
+      case functions: ShowFunctionsContext => visitShowFunctions(functions)
+      case _ => throw QueryParsingErrors.unsupportedDdlStatementForQueryError(ctx)
```

Comment:
This can't happen, and is an assert like error

Reply:
Let's remove it.